File: services/order-service/src/application/dtos/order_dtos.py
from pydantic import BaseModel, Field
from typing import List
from uuid import UUID

class OrderItemDTO(BaseModel):
    product_id: UUID
    quantity: int = Field(..., gt=0)
    unit_price: float = Field(..., gt=0)

class CreateOrderCommand(BaseModel):
    customer_id: UUID
    items: List[OrderItemDTO]

class OrderResult(BaseModel):
    order_id: UUID
    status: str
    created_at: str

File: boilerplate/java/order-service/src/main/java/com/example/orderservice/application/dtos/OrderDtos.java
package com.example.orderservice.application.dtos;

import java.util.*;
import java.time.OffsetDateTime;

public record CreateOrderCommand(UUID customerId, List<OrderItemDTO> items) {}

public record OrderItemDTO(UUID productId, int quantity, double unitPrice) {}

public record OrderResult(UUID orderId, String status, OffsetDateTime createdAt) {}
